core/Console.php

```diff
@@ -63,10 +64,14 @@ private function addCommandIfExists($command)
     {
         if (!class_exists($command)) {
             Log::warning(sprintf('Cannot add command %s, class does not exist', $command));
-        } elseif (!is_subclass_of($command, 'Piwik\Plugin\ConsoleCommand')) {
+        } else if (!is_subclass_of($command, 'Piwik\Plugin\ConsoleCommand')) {
             Log::warning(sprintf('Cannot add command %s, class does not extend Piwik\Plugin\ConsoleCommand', $command));
         } else {
-            $this->add(new $command);
+            /** @var Command $commandInstance */
+            $commandInstance = new $command;
+            if (!$this->has($commandInstance->getName())) {
+                $this->add($commandInstance);
+            }
```

Comment:
Maybe add a comment to explain that the command might be overriden in tests? Else we'll be confused by that in the future and someone might remove it.